zaza/charm_tests/ceph/tests.py

```diff
@@ -0,0 +1,89 @@
+"""Ceph-osd Testing."""
+
+import zaza.charm_tests.test_utils as test_utils
+import zaza.utilities.generic as zaza_utils
+import zaza.utilities.openstack as zaza_openstack
+
+
+class CephOsdLowLevelTest(test_utils.OpenStackBaseTest):
+    """Ceph-OSD Test Class."""
+
+    @classmethod
+    def setUpClass(cls):
+        """Run class setup for running ceph-osd tests."""
+        super(CephOsdLowLevelTest, cls).setUpClass()
+
+    def test_ceph_processes(self):
+        """Verify Ceph processes.
+
+        Verify that the expected service processes are running
+        on each ceph unit.
+        """
+        # Process name and quantity of processes to expect on each unit
+        ceph_mon_processes = {
+            'ceph-mon': 1,
+        }
+
+        ceph_osd_processes = {
+            "ceph-osd": [2, 3]
+        }
+
+        # Units with process names and PID quantities expected
+        expected_processes = {
+            "ceph-mon/0": ceph_mon_processes,
+            "ceph-mon/1": ceph_mon_processes,
+            "ceph-mon/2": ceph_mon_processes,
+            "ceph-osd/0": ceph_osd_processes,
+            "ceph-osd/1": ceph_osd_processes,
+            "ceph-osd/2": ceph_osd_processes
+        }
+
+        actual_pids = zaza_utils.get_unit_process_ids(expected_processes)
+        ret = zaza_utils.validate_unit_process_ids(expected_processes,
+                                                   actual_pids)
+        self.assertTrue(ret)
+
+    def test_services(self):
+        """Verify the ceph services.
+
+        Verify the expected services are running on the service units.
+        """
+        services = {
```

Comment:
What you in effect are doing here is `cinder` and `glance` charm tests in the `ceph` charm test.

This is one of the Amuletisms we want to avoid in Zaza.  What we could do instead is pull in relevant config or test snippets in the tests.yaml that in effect would validate that the other services are working as they should.

An example for the keystone charm which by pulling in a glance config function in effect validates that the glance services are running, that it successfully related itself to keystone etc etc:
https://review.openstack.org/#/c/589832/2/tests/tests.yaml

Reply:
Indeed. I will move that away.
I was indeed just copy/pasting the amulet tests. 
Thanks!